public/app/plugins/datasource/cloudwatch/utils/query/getStatsGroups.ts

```diff
@@ -1,41 +1,21 @@
-const antlr4 = require('antlr4');
-const ScrollQLLexer = require('./ScrollQLLexer').ScrollQLLexer;
-const ScrollQLParser = require('./ScrollQLParser').ScrollQLParser;
-const ScrollQLParserListener = require('./ScrollQLParserListener').ScrollQLParserListener;
+const byRE = /\s+by\s+/im;
+const groupsRE = /([\w@().]+)(?:(\s+as\s+)(\w+))?\s*,?\s*/iy;
```

Comment:
I would probably like a bit of explanation comment here about how this works in the sense of having a group for main term and then for alias creating 3 groups in the match you can then check and probably also about the `y` flag cause it took me a while to understand why it does not match anything if there is a `| sort @timestamp desc` for example after the stats.

Reply:
the `y` flag is not relevant to why it doesn't consume `| sort @timestamp desc`, it's just there so the regex can be run iteratively by mutating `.lastIndex`. there's simply nothing that will match `|` in the regex :). i've added some clarification for the group meanings and why it doesnt go past `|`.

Reply:
Hmm right I got it mixed with `g` flag. It also uses `lastIndex` but would basically just skip the ` | ` and match the next word. `y` treats it in a way that the match hast to start at that positions otherwise wouldn't match. So this works only with `y` (`g` would match more stuff, without flag it would match just the first group), because the regex as of itself could just match any word as most capture groups are optional.